R/ParamSet.R

```diff
@@ -397,8 +402,14 @@ ParamSet = R6Class("ParamSet",
     #'   `tolerance` arguments of `p_dbl()` and `p_int()`) . If `sanitize`
     #'   is `TRUE`, the additional effect is that, should checks pass, the
     #'   sanitized values of `xs` are added to the result as attribute `"sanitized"`.
+    #' @param presence (`character(1)`)\cr
+    #'   If `"none"` (default), no check is performed for the presence of parameters.
+    #'   If `"all"`, all parameters must be present in `xs`, except for parameters with unsatisfied dependencies.
+    #'   If `"required"`, required parameters must be present in `xs`, except for parameters with unsatisfied dependencies.
+    #'   For `"all"` and `"required"`, `TuneToken`s are not allowed to be present in `xs`.
```

Comment:
I would say (dis)allowing tune tokens is orthogonal to requiring presence, maybe we want to have an extra argument allow_tunetoken (default TRUE)? Also, do we want to forbid InternalTuneToken?

Reply:
I mean, for the sake of dependencies in `$check_dependencies` we are more liberal w/r/t TuneToken. The view there seems to be that if something is a TuneToken, the parameter is present for the sake of being depended on.

Reply:
I added `allow_token = TRUE`